hbase-server/src/main/java/org/apache/hadoop/hbase/master/cleaner/BaseTimeToLiveFileCleaner.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.cleaner;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hbase.util.EnvironmentEdgeManager;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Base class for time to live file cleaner.
+ */
+@InterfaceAudience.Private
+public abstract class BaseTimeToLiveFileCleaner extends BaseLogCleanerDelegate {
+
+  private static final Logger LOG =
+    LoggerFactory.getLogger(BaseTimeToLiveFileCleaner.class.getName());
+
+  private static final DateTimeFormatter FORMATTER =
+    DateTimeFormatter.ISO_DATE_TIME.withZone(ZoneOffset.systemDefault());
+
+  // Configured time a log can be kept after it was closed
+  private long ttlMs;
+
+  private boolean stopped = false;
+
+  @Override
+  public final void setConf(Configuration conf) {
+    super.setConf(conf);
+    this.ttlMs = getTtlMs(conf);
+  }
+
+  @Override
+  public boolean isFileDeletable(FileStatus status) {
+    // Files are validated for the second time here,
+    // if it causes a bottleneck this logic needs refactored
+    if (!valiateFilename(status.getPath())) {
```

Comment:
s/valiate/validate/

Reply:
Why a second time? Just-in-case? Can you say why we do it twice in comment? What you are afraid of?

Reply:
Just need to make sure that we are only dealing with the correct files. The cleaner chain works with 'AND', that means, if one of the cleaners return false, then we will not delete the file.

For example, we have two wal files, one is the normal one, the other is the local store one. The former one has a TTL of 1 hour, and the latter one has a TTL of 7 days. If we do not have the test here, then for all files, the latter cleaner will always return false within 7 days, and the former one will have no effect. So here we need to make sure that, each cleaner just test the file that should be checked by it.